src/actions/account.js

```diff
@@ -273,44 +281,49 @@ export const { getAccessKeys, removeAccessKey, addLedgerAccessKey, connectLedger
     SHOW_LEDGER_MODAL: null
 })
 
-export const { addAccessKey, addAccessKeySeedPhrase, clearAlert } = createActions({
+export const handleAddAccessKeySeedPhrase = (accountId, recoveryKeyPair) => async (dispatch) => {
+    await dispatch(addAccessKeySeedPhrase(accountId, recoveryKeyPair))
+    dispatch(redirectTo('/profile'))
+}
+
+export const handleCreateAccountWithSeedPhrase = (accountId, recoveryKeyPair, fundingContract, fundingKey) => async (dispatch) => {
+    await dispatch(createAccountWithSeedPhrase(accountId, recoveryKeyPair, fundingContract, fundingKey))
+    const account = await wallet.refreshAccount()
+    const promptTwoFactor = await wallet.twoFactor.checkCanEnableTwoFactor(account)
+
+    if (fundingContract && promptTwoFactor) {
+        dispatch(redirectTo('/enable-two-factor'))
```

Comment:
@marcinbodnar @vgrichina  whenever I'm creating a **new** account w/ seed phrase or link and then dispatch a redirect, a REPLACE action gets dispatched right after redirect, in this case, replacing `/enable-two-factor` with `/`. I can't seem to figure out why this happens. Do you have any idea?

Note that it only happens while creating a new account and then redirecting.

![Screen Shot 2020-09-17 at 6 12 18 PM](https://user-images.githubusercontent.com/24921205/93549156-91943b00-f91d-11ea-9925-7258461bc8ce.png)

Reply:
@Patrick1904 
The problem is in `handleCreateAccountWithSeedPhrase`, instead of `await wallet.refreshAccount()` you need to dispatch `refreshAccount` action because now when you redirect two lines later the account is not visible in Redux.